ext/mysqlnd/mysqlnd_driver.c

```diff
@@ -87,6 +87,11 @@ PHPAPI void mysqlnd_library_init(void)
 		mysqlnd_register_builtin_authentication_plugins();
 
 		mysqlnd_reverse_api_init();
+
+#if MYSQLND_CHARSETS_SANITY_CHECK == 1
```

Comment:
This seems fine, but I assume we have no way of incorporating this into the build process, right? It can only be run manually by the developer.

Reply:
Yeah I don't think we can incorporate this, it would also be very slow to do because it does exhaustive testing.
We have something similar with mbstring where there's an optional sanity check too that the developer has to manually enable.